.ci/upload-all-jar.sh

```diff
@@ -28,8 +28,8 @@ RELEASE_ID=$(curl -s -X GET \
 
 
 curl -i -H "Authorization: token $GITHUB_TOKEN" \
-  -H "Content-Type: application/zip" \
-  --data-binary @"target/checkout/target/checkstyle-$TARGET_VERSION-all.jar" \
+  -H "Content-Type: application/java-archive" \
```

Comment:
[According to MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types), the proper type should be `application/java-archive` if we want to be exact.